SYCL/ESIMD/matrix_transpose_usm.cpp

```diff
@@ -241,12 +241,10 @@ ESIMD_INLINE void transpose16(int *buf, int MZ, int block_col, int block_row) {
   }
 }
 
-bool runTest(unsigned MZ, unsigned block_size) {
-  queue q(esimd_test::ESIMDSelector{}, esimd_test::createExceptionHandler(),
-          property::queue::enable_profiling{});
+bool runTest(queue &q, unsigned MZ, unsigned block_size) {
   auto dev = q.get_device();
   auto ctxt = q.get_context();
-  int *M = static_cast<int *>(malloc_shared(MZ * MZ * sizeof(int), dev, ctxt));
+  int *M = malloc_shared<int>(MZ * MZ, dev, ctxt);
```

Comment:
since you are improving the test anyway, would be good to have sycl::free as well. Also malloc_shared can accept q instead if dev, ctx pair

Reply:
Hmm, sycl::free is there, see the lines 305 and 326.

Reply:
> Hmm, sycl::free is there, see the lines 305 and 326.

Oh, missed that, sorry.